assets/src/editor/events/ticket-assignments-manager/ticket-assignments-manager-modal.js

```diff
@@ -41,463 +41,180 @@ const {
-			assigned,
-			removed
-		);
-		const {
-			currentlyAssigned,
-			canRemoveAssignment,
-		} = this.getActionModifiers(
-			hasTicket,
-			isAssigned,
-			isRemoved,
-			totalTicketAssignmentsForDate,
-			totalDateAssignmentsForTicket
-		);
-		const action = this.getAction(
-			currentlyAssigned,
-			canRemoveAssignment,
-			totalDateAssignmentsForTicket
-		);
-		const assignmentsErrorClass = this.getAssignmentsErrorClass(
-			totalTicketAssignmentsForDate,
-			totalDateAssignmentsForTicket
-		);
-		const { icon, bgColor } = this.getIconAndBgColor(
-			hasTicket,
-			isAssigned,
-			isRemoved
-		);
-		return {
-			type: 'cell',
-			class: `ee-tam-date-row-ticket${ assignmentsErrorClass }`,
-			value: (
-				<IconButton
-					icon={ icon }
-					className={ bgColor }
-					size={ 45 }
-					onClick={ ( event ) => {
-						event.preventDefault();
-						event.stopPropagation();
-						action( dateEntity, ticketEntity );
-					} }
-					onKeyDown={ ( event ) => {
-						if ( event.keyCode === ENTER ) {
-							event.preventDefault();
-							event.stopPropagation();
-							action( dateEntity, ticketEntity );
-						}
-					} }
-				/>
-			),
-		};
-	};
-
-	/**
-	 * not only do we need to know how many existing relations between tickets
-	 * and dates there are, but also how many are queued for removal as well
-	 * as new ones queued for being added
-	 *
-	 * @function
-	 * @param {Object} dateEntity
-	 * @param {Object} ticketEntity
-	 * @param {Object} assigned
-	 * @param {Object} removed
-	 * @return {Object} JSON object with ticket and date assignment counts
-	 */
-	calculateTotalAssignmentCounts = ( dateEntity, ticketEntity, assigned, removed ) => {
-		const ticketCountForDate = this.assignmentCounts.dateEntities[ dateEntity.id ];
-		const dateCountForTicket = this.assignmentCounts.ticketEntities[ ticketEntity.id ];
-		const assignedTickets = handler.assignedCount( assigned, null, ticketEntity );
-		const removedTickets = handler.removedCount( removed, null, ticketEntity );
-		const assignedDates = handler.assignedCount( assigned, dateEntity );
-		const removedDates = handler.removedCount( removed, dateEntity );
-		const totalTicketAssignmentsForDate = ticketCountForDate +
-			assignedDates - removedDates;
-		const totalDateAssignmentsForTicket = dateCountForTicket +
-			assignedTickets - removedTickets;
-		return {
-			totalTicketAssignmentsForDate,
-			totalDateAssignmentsForTicket,
-		};
-	};
+					class: 'ee-tam-year-row',
+				},
+				{
+					type: 'cell',
+					value: year,
+					class: 'ee-tam-date-label',
+				},
+			];
+			ticketEntitiesForYear.forEach( () => {
+				rowData.push(
+					{
+						type: 'cell',
+						value: '',
+						render: ( rowNumber, colNumber ) => (
+							<td
+								key={ `row-${ rowNumber }-col-${ colNumber }` }
+								className={
+									'ee-tam-date-row-ticket ee-rTable-body-td'
+								}
+							>
+							</td>
+						),
+					}
+				);
+			} );
+			return rowData;
+		},
+		[]
```

Comment:
`yearRow` does not need any dependencies because it is not using anything from outside its scope (but only what is passed in on the callback).